### PR TITLE
Remove non-triggered code, increase test coverage

### PR DESCRIPTION
**Pull request by berland**
These corner-cases have been fixed already in surrounding code,
lets elevate remaining corner cases to bugs.

### CODE REVIEW

File: pyscal/gasoil.py [hunk 1]
```diff
@@ -338,8 +338,6 @@ def add_fromtable(
                 self.sgro = sgro_estimate
         if pccolname in dframe:
             # Incoming dataframe must cover the range:
-            if dframe[sgcolname].min() > self.table["SG"].min():
```
Comment (berland) [inline pyscal/gasoil.py:348]: (this is never reached because there is another test further up that catches it)

File: pyscal/gasoil.py [hunk 2]
```diff
@@ -851,28 +842,10 @@ def slgof_df(self) -> pd.DataFrame:
...
-            else:
-                # Give up repairing the table:
-                logger.critical(
-                    "SLGOF does not start at the correct value. Please report as bug."
-                )
-                logger.error("slgof_sl_mismatch: %f", slgof_sl_mismatch)
-                logger.error(str(slgof.head()))
+        assert (
+            slgof_sl_mismatch < epsilon
+        ), f"Bug: slgof_sl_mismatch was {slgof_sl_mismatch}, too large"
```
Comment (jondequinor) [inline pyscal/gasoil.py:848]: Same here, really. You can run `python -O -m pyscal` and ignore all `assert`. Is that what you want?

File: pyscal/gasoil.py [hunk 3]
```diff
@@ -167,16 +167,9 @@ def __init__(
...
-            self.table.loc[len(self.table) + 1, "SG"] = 1 - self.swl
-            self.table.sort_values(by="SG", inplace=True)
-        # Ensure the value closest to 1-swl is actually 1-swl:
-        swl_right_index = (
-            (self.table["SG"] - (1 - self.swl)).abs().sort_values().index[0]
-        )
-        self.table.loc[swl_right_index, "SG"] = 1 - self.swl
+        assert np.isclose(
+            self.table["SG"].max(), 1 - self.swl
+        ), "Bug, max(sg) was not close to 1-swl"
```
[the remark this answers is not in the document]
Comment (berland) [reply pyscal/gasoil.py:172]: This is test coverage cheating. After searching for ways to trigger the ValueError in test code, I gave up and changed it into an assert, which makes it into one covered line, instead of one covered and one missed. It should be ok if asserts are ignored by command line options, it will crash somewhere anyway. If someone is able to get into this situation, we want it reported as a bug nonetheless.

[the remark this answers is not in the document]
Comment (jondequinor) [reply pyscal/gasoil.py:172]: > This is test coverage cheating
Then by Goodhart's law you should not use coverage, or in this case ignore it.

>  it will crash somewhere anyway
No, it should fail as soon as possible.

> After searching for ways to trigger the ValueError in test code
Mock `np.isclose` if you really need to?

[the remark this answers is not in the document]
Comment (berland) [reply pyscal/gasoil.py:172]: I will follow Goodheart from now on :) 

In this particular case, independent of Goodheart, I will remove the test entirely. It should be impossible to reach by construction, and sufficiently proven by pytest-hypothesis.

File: pyscal/gasoil.py (hunk 2 above)
Comment (berland) [reply pyscal/gasoil.py:848]: Changed to `if` and `ValueError`.

**Review by jondequinor — APPROVED** [review]
I bit hard to read this diff, but (removing business code and) adding tests is always good.

Thanks
